lib/src/track/local/video.dart

```diff
@@ -239,5 +265,207 @@ extension LocalVideoTrackExt on LocalVideoTrack {
     await restartTrack(
       options.copyWith(deviceId: deviceId),
     );
+
+    await replaceTrackForMultiCodecSimulcast(mediaStreamTrack);
+  }
+
+  Future<void> replaceTrackForMultiCodecSimulcast(
+      rtc.MediaStreamTrack newTrack) async {
+    simulcastCodecs.forEach((key, simulcastTrack) async {
+      await simulcastTrack.sender?.replaceTrack(newTrack);
+      simulcastTrack.mediaStreamTrack = mediaStreamTrack;
+    });
+  }
+
+  Future<List<String>> setPublishingCodecs(List<lk_rtc.SubscribedCodec> codecs,
+      LocalTrackPublication publication) async {
+    logger.fine('setPublishingCodecs $codecs');
+
+    // only enable simulcast codec for preference codec setted
+    if (codec == null && codecs.isNotEmpty) {
```

Comment:
I must be missing something obvious, where is `codec` defined here or is it just always `null` ?

Reply:
dart analyzer does not recommend using `this.`, there is no variable named `codec` in this function scope, xD

<img width="483" alt="image" src="https://github.com/livekit/client-sdk-flutter/assets/17733746/3c2151a0-3fc5-42dc-9fac-64f28e8d3a63">